actionview/test/template/lookup_context_test.rb

```diff
@@ -65,16 +65,29 @@ def teardown
     assert_equal Mime::SET.to_a, @lookup_context.formats
   end
 
+  test "ignores nil format" do
+    @lookup_context.formats = [:html, nil, :text]
+    assert_equal [:html, :text], @lookup_context.formats
+  end
+
   test "handles explicitly defined */* formats fallback to :js" do
     @lookup_context.formats = [:js, Mime::ALL]
-    assert_equal [:js, *Mime::SET.symbols], @lookup_context.formats
+    assert_equal [:js, *Mime::SET.symbols].uniq, @lookup_context.formats
   end
 
   test "adds :html fallback to :js formats" do
     @lookup_context.formats = [:js]
     assert_equal [:js, :html], @lookup_context.formats
   end
 
+  test "raises on invalid format assignment" do
+    ex = assert_raises ArgumentError do
+      @lookup_context.formats = [:html, :javascript]
```

Comment:
I guess it's not too obtuse when the right JavaScript format `:js` is just above, but should we consider saying `:invalid` instead? Should we pass 2 so we see the comma in the message?